src/coreclr/src/pal/src/libunwind/CMakeLists.txt

```diff
@@ -1,21 +1,125 @@
-add_subdirectory(src)
-
 # define variables for the configure_file below
 
-if (CLR_CMAKE_HOST_ARCH_AMD64)
-    set(arch x86_64)
-elseif(CLR_CMAKE_HOST_ARCH_ARM64)
-    set(arch aarch64)
-elseif(CLR_CMAKE_PLATFROM_ARCH_ARM)
-    set(arch arm)
-elseif(CLR_CMAKE_HOST_ARCH_I386)
-    set(arch x86)
-endif()
-
 set(PKG_MAJOR "1")
 set(PKG_MINOR "5")
 set(PKG_EXTRA "-rc2")
+if(CLR_CMAKE_HOST_UNIX)
+    if (CLR_CMAKE_HOST_ARCH_AMD64)
+      set(arch x86_64)
+    elseif(CLR_CMAKE_HOST_ARCH_ARM64)
+      set(arch aarch64)
+    elseif(CLR_CMAKE_PLATFROM_ARCH_ARM)
+      set(arch arm)
+    elseif(CLR_CMAKE_HOST_ARCH_I386)
+        set(arch x86)
+    endif ()
+endif(CLR_CMAKE_HOST_UNIX)
+
+if(CLR_CMAKE_HOST_WIN32)
+    if (CLR_CMAKE_TARGET_ARCH_AMD64)
+      set(TARGET_AMD64 1)
+      set(arch x86_64)
+      add_definitions(-D__x86_64__)
+      add_definitions(-D__amd64__)
+    elseif(CLR_CMAKE_TARGET_ARCH_ARM64)
+      set(TARGET_AARCH64 1)
+      set(arch aarch64)
+      add_definitions(-D__aarch64__)
+    elseif(CLR_CMAKE_TARGET_ARCH_ARM)
+      set(TARGET_ARM 1)
+      set(arch arm)
+      add_definitions(-D__arm__)
+    else ()
+      message(FATAL_ERROR "Unrecognize value in environment variable TARGET")
+    endif ()
+
+    set(PACKAGE_STRING "libunwind")
+    set(PACKAGE_BUGREPORT "")
+
+    include(CheckCSourceCompiles)
+    include(CheckIncludeFiles)
+
+    # Our posix abstraction layer will provide these headers
+    set(HAVE_ELF_H 1)
+    set(HAVE_ENDIAN_H 1)
+
+    # MSVC compiler is currently missing C11 stdalign.h header
+    # Fake it until support is added
+    check_include_files(stdalign.h HAVE_STDALIGN_H)
+    if (NOT HAVE_STDALIGN_H)
+        configure_file(include/win/fakestdalign.h.in ${CMAKE_CURRENT_BINARY_DIR}/include/stdalign.h)
```

Comment:
This can be just 
`file(COPY include/win/fakestdalign.h DESTINATION ${CMAKE_CURRENT_BINARY_DIR}/include/stdalign.h)` 
as the files don't need any configuration.

Reply:
I would prefer
```cmake
configure_file(include/win/fakestdalign.h.in
               ${CMAKE_CURRENT_BINARY_DIR}/include/stdalign.h 
               COPYONLY)
```